cmd/nvidia-ctk-installer/container/runtime/containerd/config_v2_test.go

```diff
@@ -30,51 +30,136 @@ import (
+//   - Existing config without nvidia entries (TestUpdateV2Config_ExistingConfigWithoutNvidia)
+//   - Existing config with nvidia entries (TestUpdateV2Config_ExistingConfigWithNvidia)
+//
+// Each test case represents a specific permutation of configuration options:
+//   - Runtime name: "nvidia" (standard) or custom names
+//   - CDI enablement: enabled or disabled
+//   - Default runtime: whether to set the nvidia runtime as default
+//   - Runc inheritance: whether runc runtime exists to inherit settings from
+//
+// This approach ensures consistent test coverage across all scenarios and makes
+// it easy to verify that the same feature works correctly regardless of the
+// initial state of the containerd configuration.
+func getSharedTestCases() []testCase {
+	return []testCase{
+		{
+			name:         "nvidia runtime without CDI, not default, no runc",
+			runtimeName:  "nvidia",
+			enableCDI:    false,
+			setAsDefault: false,
+			withRunc:     false,
+		},
 		{
-			setAsDefault:               false,
-			runtimeName:                "nvidia",
-			expectedDefaultRuntimeName: nil,
+			name:         "nvidia runtime with CDI enabled, not default, no runc",
+			runtimeName:  "nvidia",
+			enableCDI:    true,
+			setAsDefault: false,
+			withRunc:     false,
 		},
 		{
-			setAsDefault:               false,
-			runtimeName:                "NAME",
-			expectedDefaultRuntimeName: nil,
+			name:         "nvidia runtime without CDI, set as default, no runc",
+			runtimeName:  "nvidia",
+			enableCDI:    false,
+			setAsDefault: true,
+			withRunc:     false,
 		},
 		{
-			setAsDefault:               true,
-			runtimeName:                "nvidia",
-			expectedDefaultRuntimeName: "nvidia",
+			name:         "nvidia runtime with CDI enabled, set as default, no runc",
+			runtimeName:  "nvidia",
+			enableCDI:    true,
+			setAsDefault: true,
+			withRunc:     false,
 		},
 		{
-			setAsDefault:               true,
-			runtimeName:                "NAME",
-			expectedDefaultRuntimeName: "NAME",
+			name:         "custom runtime (CUSTOM) without CDI, not default, no runc",
+			runtimeName:  "CUSTOM",
+			enableCDI:    false,
+			setAsDefault: false,
+			withRunc:     false,
+		},
+		{
+			name:         "custom runtime (CUSTOM) with CDI enabled, set as default, no runc",
+			runtimeName:  "CUSTOM",
+			enableCDI:    true,
+			setAsDefault: true,
+			withRunc:     false,
+		},
+		{
+			name:         "nvidia runtime without CDI, not default, with runc",
+			runtimeName:  "nvidia",
+			enableCDI:    false,
+			setAsDefault: false,
+			withRunc:     true,
+		},
+		{
+			name:         "nvidia runtime with CDI enabled, set as default, with runc",
+			runtimeName:  "nvidia",
+			enableCDI:    true,
+			setAsDefault: true,
+			withRunc:     true,
 		},
 	}
+}
+
+// TestUpdateV2Config_NoConfigFile tests the scenario when there is no
+// containerd config file present
+func TestUpdateV2Config_NoConfigFile(t *testing.T) {
```

Comment:
As discussed:
* offline in our call yesterday
* https://github.com/NVIDIA/nvidia-container-toolkit/pull/1272#discussion_r2309632391
* https://github.com/NVIDIA/nvidia-container-toolkit/pull/1280/files#diff-743eab44a71e6e333fef0be742b20bcdb7188341b5b8e6542bc94ec64704de35R35

I think these tests are testing the wrong thing. Although we may want to consider updating the tests for `Update` config, what we really want is tests to ensure the correct behaviour of the `Setup`->`Cleanup` cycle.

Would adapting the proposed implementation from the last two points above be feasible?